test/functional/cmdLineTests/criu/src/org/openj9/criu/TimeChangeTest.java

```diff
@@ -0,0 +1,49 @@
+/*******************************************************************************
+ * Copyright (c) 2022, 2022 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package org.openj9.criu;
+
+import java.nio.file.Paths;
+import java.nio.file.Path;
+
+public class TimeChangeTest {
+
+	// The elapsed time is expected less than 2 second.
+	private static long MAX_ELAPSED_TIME = 2000 * 1000 * 1000;
+	private static Path imagePath = Paths.get("cpData");
+
+	public static void main(String args[]) {
+		new TimeChangeTest().testSystemNanoTime();
+	}
+
+	public void testSystemNanoTime() {
+		final long beforeCheckpoint = System.nanoTime();
+		System.out.println("System.nanoTime() before CRIU checkpoint: " + beforeCheckpoint);
+		CRIUTestUtils.checkPointJVM(imagePath);
+		final long afterRestore = System.nanoTime();
+		final long elapsedTime = afterRestore - beforeCheckpoint;
+		if (elapsedTime < MAX_ELAPSED_TIME) {
```

Comment:
instead of `MAX_ELAPSED_TIME` you could use actual elapsed time based on wall clock (System.currentTimeMillis) and ensure that the nanoTime elapsed is less

Reply:
The `MAX_ELAPSED_TIME` is a rough estimate for a checkpoint process. I feel the wall clock time diff is too loose after first run (total three runs w/ 2 second apart).